pkg/gui/controllers/toggle_whitespace_action.go

```diff
@@ -1,21 +1,29 @@
 package controllers
 
 import (
+	"github.com/jesseduffield/lazygit/pkg/gui/context"
 	"github.com/jesseduffield/lazygit/pkg/gui/types"
+	"github.com/samber/lo"
 )
 
 type ToggleWhitespaceAction struct {
 	c *ControllerCommon
 }
 
 func (self *ToggleWhitespaceAction) Call() error {
-	self.c.State().SetIgnoreWhitespaceInDiffView(!self.c.State().GetIgnoreWhitespaceInDiffView())
+	contextsThatDontSupportIgnoringWhitespace := []types.ContextKey{
+		context.STAGING_MAIN_CONTEXT_KEY,
+		context.STAGING_SECONDARY_CONTEXT_KEY,
+		context.PATCH_BUILDING_MAIN_CONTEXT_KEY,
+	}
 
-	toastMessage := self.c.Tr.ShowingWhitespaceInDiffView
-	if self.c.State().GetIgnoreWhitespaceInDiffView() {
-		toastMessage = self.c.Tr.IgnoringWhitespaceInDiffView
+	if lo.Contains(contextsThatDontSupportIgnoringWhitespace, self.c.CurrentContext().GetKey()) {
+		// Ignoring whitespace is not supported in these views. Let the user
+		// know that it's not going to work in case they try to turn it on.
+		return self.c.ErrorMsg(self.c.Tr.IgnoreWhitespaceNotSupportedHere)
```

Comment:
I've pushed a change here after testing locally: I thought that the error message wasn't appearing cos of some bug but in fact it was the case that ignore-whitespace was enabled and so I couldn't _disable_ it meaning I was spamming the keybinding but didn't see a change. So I've changed it to just show the error regardless of whether you're toggling on/off, which I think is fine.